<?xml version="1.0"?>
<launch>

    <!-- Desired interrobot distance -->
    <param name = "inter_distance_x" value="10"/>
    <param name = "inter_distance_y" value="0"/>

    <param name = "formation" value="line"/>

    <!-- Launch n robots -->
    <node name="robot_1" pkg="mr_rendezvous_deployment" type="naive_robot.py" output="screen">
        <param name = "robot_id" value = "1" />
        <param name = "position" value = "2.5 7.5"/>
        <param name = "neighbors" value = "2" type="string"/>
        <param name = "t_local" value = "0.5"/>
    </node>
    <node name="robot_2" pkg="mr_rendezvous_deployment" type="naive_robot.py" output="screen">
        <param name = "robot_id" value = "2" />
        <param name = "position" value = "0 0"/>
        <param name = "neighbors" value = "1 3" type="string"/>
        <param name = "t_local" value = "0.5"/>
    </node>
    <node name="robot_3" pkg="mr_rendezvous_deployment" type="naive_robot.py" output="screen">
        <param name = "robot_id" value = "3" />
        <param name = "position" value = "-2.5 -5.0"/>
        <param name = "neighbors" value = "2 4" type="string"/>
        <param name = "t_local" value = "0.5"/>
    </node>
    <node name="robot_4" pkg="mr_rendezvous_deployment" type="naive_robot.py" output="screen">
        <param name = "robot_id" value = "4" />
        <param name = "position" value = "-7.5 -5.0"/>
        <param name = "neighbors" value = "3" type="string"/>
        <param name = "t_local" value = "0.5"/>
    </node>
    
    <!-- Launch visualization -->
    <node name="position_plotter" pkg="mr_rendezvous_deployment" type="visualization_node.py" output="screen" args="4"/>

</launch>